helios-services/src/main/java/com/spotify/helios/master/reaper/DeadAgentReaper.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2016 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.master.reaper;
+
+import com.spotify.helios.common.Clock;
+import com.spotify.helios.common.SystemClock;
+import com.spotify.helios.common.descriptors.AgentInfo;
+import com.spotify.helios.common.descriptors.HostStatus;
+import com.spotify.helios.master.MasterModel;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.lang.time.DurationFormatUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Random;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * De-registers dead agents. An agent that has been DOWN for more than X hours is considered
+ * dead.
+ */
+public class DeadAgentReaper extends RateLimitedService<String> {
+
+  private static final double PERMITS_PER_SECOND = 0.2; // one permit every 5 seconds
+  private static final Clock SYSTEM_CLOCK = new SystemClock();
+  private static final int DELAY = 30;
+  private static final TimeUnit TIME_UNIT = TimeUnit.MINUTES;
```

Comment:
Why is this needed? If you want to store a type of time unit with a value, `Duration` is probably a better choice.